doc/mgmt/SONiC_Design_Doc_Unified_FRR_Mgmt_Interface.md

```diff
@@ -0,0 +1,1069 @@
+set_ext_community_inline = 1*64VCHAR ; Set BGP extended community attribute inline
+set_ext_community_ref    = 1*64VCHAR ; Refer BGP extended community attribute from extended community table
+```
+#### 3.2.1.8 ROUTE_REDISTRIBUTE
+```JSON
+;Defines route redistribution table
+;
+;Status: stable
+
+key                      = ROUTE_REDISTRIBUTE|vrf|src_protocol|dst_protocol|addr_family ;
+vrf                      = 1\*15VCHAR ; VRF name
+src_protocol             = "connected" / "static" / "ospf" / "ospf3"
+dst_protocol             = "bgp"
+addr_family              = "ipv4" / "ipv6"   
+route_map                = 1*64VCHAR ; route map filter to apply for redistribution
+```
+
+### 3.2.1.9 IP_PREFIX_SET
+```JSON
+;Defines prefix set table
+;
+;Status: stable
+
+key           = IP_PREFIX_SET:name          ; prefix_set_name must be unique
+name          = 1*255VCHAR ; community set name  
+mode          = "IPv4"/"IPv6" ; mode of prefix set.
+
+````
+#### 3.2.1.9.1 IP_PREFIX
+```JSON
+;Defines prefix table
+;
+;Status: stable
+key              = IP_PREFIX:set_name:ip_prefix:masklength_range; an instance of this key will be repeated for each prefix
+                                                             ;  an instance of this key/value pair will be repeated for each prefix
+set_name         = 1*255VCHAR ; community set name                                                             
+ip_prefix        = IPv4prefix / IPv6prefix   ; prefix, example 1.1.1.1/32              
+masklength_range = 1*255VCHAR                ; exact or (masklength_range..low-masklength_range_high). example 8..16 or exact
+```
+### 3.2.1.10 BGP_COMMUNITY_SET
+```JSON
+;Defines community table
+;
+;Status: stable
+key              = BGP_COMMUNITY_SET|name  ; name must be unique
+set_type          = "STANDARD"/"EXPANDED"
+match_action      = "ANY/ALL"
+community_member = string list ; community member list
+                               ; Acceptable List of communities as ("AA:NN","local-AS", "no-advertise", "no-export" | regex)
+
+````
+### 3.2.1.11 BGP_EXT_COMMUNITY_SET
+```JSON
+;Defines extended community table
+;
+;Status: stable
+key               = BGP_EXT_COMMUNITY_SET|name          ; name must be unique
+set_type          = "STANDARD"/"EXPANDED"
+match_action      = "ANY/ALL"
+community_member = string list; community member list
+                              ; Acceptable List of communities as ("route-target/route-origin:AA:NN" or "IP_Address" or regex)
+````
+### 3.2.1.12 BGP_AS_PATH_SET
+```JSON
+;Defines extended community table
+;
+;Status: stable
+key           = AS_PATH_SET|name          ; name must be unique
+
+as_path_member = string list; AS path list
+                            ;Acceptable List of as paths "string, string"
+````
+
+### 3.2.2 APP DB
+N/A
+
+### 3.2.3 STATE DB
+No changes to State DB, State and statistics information will be retrieved directly from FRR-BGP.
```

Comment:
I'd suggest to use db as a source.
otherwise someone could run the requests many times and dos frr

Reply:
Yes, our long term goal is to push FRR contents into state DB and read them from mgmt-framework container, but for now, we fetch information directly from FRR BGP.

Reply:
I don't think it is good idea.

Reply:
Yes, we're aware of this, since this needs an FRR enhancement to push state information to state DB, will take this up later.